pkg/trace/agent/agent.go

```diff
@@ -422,6 +426,21 @@ func traceContainsError(trace pb.Trace) bool {
 	return false
 }
 
+func filteredByTags(root *pb.Span, require, reject []*config.Tag) bool {
+	for _, tag := range reject {
+		if v, ok := root.Meta[tag.K]; ok && (tag.V == "" || (v != "" && v == tag.V)) {
```

Comment:
I updated this logic to cover all edge cases. (Initially failed for funky cases e.g. if reject/require tag has key and value but Meta only has value)

Reply:
Ok, I see. I did not consider this use case. Here, you are considering having set the key _only_ to mean any value. I guess that's fine.
```suggestion
		if v, ok := root.Meta[tag.K]; ok && (tag.V == "" || v == tag.V) {
```
Should be good enough right?